src/Diamante/DeskBundle/Infrastructure/Attachment/FileStorage/LocalFileStorageService.php

```diff
@@ -69,15 +70,27 @@ public function remove($filename)
         $this->fs->remove($this->uploadDir->getRealPath() . '/' . $filename);
     }
 
-    public static function create($attachmentUploadDirPath, $fs)
+    public static function create($kernelProjectDir, Filesystem $fs)
     {
+        $attachmentUploadDirPath = realpath($kernelProjectDir)
+            . DIRECTORY_SEPARATOR
+            . 'public'
+            . DIRECTORY_SEPARATOR
+            . Attachment::PATH_TO_ATTACH_DIR;
+        if (!$fs->exists($attachmentUploadDirPath)) {
+            $fs->mkdir($attachmentUploadDirPath);
+        }
```

Comment:
Attachments directory should be created before installation, there is also a check for this
`https://github.com/eltrino/diamantedesk-application/blob/oro-3.1-upgrade/var/DiamanteDeskRequirements.php:20`
Also, note that we have attachment directory which is used by Oro and attachments for Diamante purposes.

Reply:
As I can see in this file we check '/var/attachments' directory, not '/src/attachments'. And what about if we have already installed application?